core/src/main/java/com/linecorp/armeria/common/ThreadLocalRequestContextStorage.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import static com.linecorp.armeria.internal.common.RequestContextUtil.newIllegalContextPoppingException;
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.Nullable;
+
+import io.netty.util.concurrent.FastThreadLocal;
+import io.netty.util.internal.InternalThreadLocalMap;
+
+final class ThreadLocalRequestContextStorage implements RequestContextStorage {
+
+    private static final FastThreadLocal<RequestContext> context = new FastThreadLocal<>();
+
+    static final ThreadLocalRequestContextStorage INSTANCE = new ThreadLocalRequestContextStorage();
```

Comment:
Think you could use enum Singleton pattern if you want.

Reply:
Ah, will do. 😄